### ISSUE TITLE
Magda Integration with NationalMap V8

### ISSUE THREAD

**Issue by t83714** [TerriaJS/nationalmap#981]
### Magda Integration with NationalMap V8

This is now waiting on:
- [ ] Nationalmap v8 release
- [ ] PR & release on DGA magda

------

Original ticket:

There are a few things we might want to check in order to make sure Magda still works well with NationalMap V8:
- [x] Magda currently sends Catalogue config to National Map via [postMessage approach](https://docs.terria.io/guide/deploying/controlling-in-an-iframe-or-popup/) and we are sending a `magda-item` type catalog item to nationalMap see [here](https://github.com/magda-io/magda/blob/74f100695b88d9498bdf11ae723ad0a97d566222/magda-web-client/src/Components/Common/DataPreviewMapOpenInNationalMapButton.js#L29). We need to check:
  - Does the [postMessage approach](https://docs.terria.io/guide/deploying/controlling-in-an-iframe-or-popup/) still work
  - Does the current config data sent over works with v8
    - As v8 use [MagdaReference.ts](https://github.com/TerriaJS/terriajs/blob/next/lib/Models/MagdaReference.ts) to handle `magda-item`, we want to make sure its functionality is on a par with previous [MagdaCatalogItem.js](https://github.com/TerriaJS/terriajs/blob/master/lib/Models/MagdaCatalogItem.js)
  - I think it's a `high priority` item to check
- [ ] As Magda now can host data files in its internal storage and some of data files might be protected, to make Magda Catalogue item still works with those internal data files, Terria’s ajax requests need to include cookie headers (assuming users already logged in to Magda). If we use xhr, we need to set [withCredentials](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/withCredentials) to true. If we use fetch api, we need to set [credentials](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch) to `include`.  Magda Catalogue Item s are previously handled by [MagdaCatalogItem.js](https://github.com/TerriaJS/terriajs/blob/master/lib/Models/MagdaCatalogItem.js) but in v8, it's [MagdaReference.ts](https://github.com/TerriaJS/terriajs/blob/next/lib/Models/MagdaReference.ts). 
  - I think it's a `low priority` item to solve if it's too hard as I can hide the `Open in National Map Button` for internal data files
- [ ] As Magda now can host data files in its internal storage, I have upgraded the [MagdaCatalogItem.js](https://github.com/magda-io/magda-preview-map/commit/ffb334619f2618ef88d0ae836eaf8d2572db69b5) that is used by Magda's [Map Preview Module](https://github.com/magda-io/magda-preview-map). We probably want to migrate the similar changes to [MagdaReference.ts](https://github.com/TerriaJS/terriajs/blob/next/lib/Models/MagdaReference.ts).
  - I think it's a `low priority` item to solve and only need to solve if we solve the item above

**Comment by AnaBelgun** [comment]
@mwu2018 The integration needs to be checked both ways: 
- data (on NationalMap) should be visualised on data.gov.au if geospatial 
- data in data.gov.au should be harvested in NationalMap; currently there might be discrepancies between the whole geospatial data catalogue in data.gov.au and what is available in NationalMap v7 - TBC

**Comment by t83714** [comment]
https://nationalmap.dev.saas.terria.io/

**Comment by t83714** [comment]
Source Code Open National Map Button: 
https://github.com/magda-io/magda/blob/2dd76a04bf3d6ff7261c14531098e8ac9e32ca21/magda-web-client/src/Components/Common/DataPreviewMapOpenInNationalMapButton.js#L65

**Comment by mwu2018** [comment]
> @mwu2018 The integration needs to be checked both ways:
> 
> * data (on NationalMap) should be visualised on data.gov.au if geospatial
> * data in data.gov.au should be harvested in NationalMap; currently there might be discrepancies between the whole geospatial data catalogue in data.gov.au and what is available in NationalMap v7 - TBC

On the second point, because the national map queries ckan with a set of filters, it is quite normal that the number of ckan items will be different from that in data.gov.au.

See below:
```
Jacky Jiang  2 hours ago
If it’s that one, it pull directly from DGA ckan using this query:
fq: +(res_format:(geojson OR GeoJSON OR kml OR KML OR kmz OR KMZ OR wms OR WMS OR CSV-GEO-AU OR csv-geo-au OR aus-geo-csv OR "Esri REST")) -(organization:alpine-shire-council OR organization:brimbank-city-council OR organization:brisbane-city-council OR organization:bundaberg-regional-council OR organization:cairns-regional-council OR organization:cardinia-shire-council OR organization:central-goldfields-shire-council OR organization:city-of-ballarat OR organization:city-of-boroondara OR organization:city-of-casey OR organization:city-of-darwin OR organization:city-of-gold-coast OR organization:city-of-greater-bendigo OR organization:city-of-greater-dandenong OR organization:city-of-greater-geelong OR organization:city-of-hobart OR organization:city-of-launceston OR organization:cityofmelbourne OR organization:city-of-monash OR organization:city-of-palmerston OR organization:city-of-port-phillip OR organization:city-of-stonnington OR organization:city-of-whittlesea OR organization:city-of-yarra OR organization:colac-otway-shire OR organization:corangamite-shire-council OR organization:frankston-city-council OR organization:fraser-coast-regional-council OR organization:glen-eira-city-council OR organization:glenelg-shire-council OR organization:gcc OR organization:golden-plains-shire-council OR organization:greater-shepparton-city-council OR organization:hindmarsh-shire-council OR organization:hobsons-bay-city-council OR organization:horsham-rural-city-council OR organization:city-of-ipswich OR organization:lake-macquarie-city-council OR organization:latrobe-city-council OR organization:loddon-shire-council OR organization:logan-city-council OR organization:macedon-ranges-shire-council OR organization:manningham-city-council OR organization:mansfield-shire-council OR organization:maribyrnong-city-council OR organization:melton-city-council OR organization:mildura-rural-city-council OR organization:www-mitchellshirecouncil-vic-gov-au OR organization:moorabool-shire-council OR organization:moreton-bay-regional-council OR organization:mosmanmunicipalcouncil OR organization:nillumbik-shire-council OR organization:noosa-shire-council OR organization:northern-grampians-shire-council OR organization:southern-grampians-shire-council OR organization:strathbogie-shire-council OR organization:surf-coast-shire-council OR organization:swan-hill-rural-city-council OR organization:toowoomba-regional-council OR organization:townsville-city-council OR organization:wangaratta-rural-city-council OR organization:warrnambool-city-council OR organization:west-wimmera-shire-council OR organization:wodonga-city-council OR organization:wyndham-city-council OR organization:yarra-ranges-council OR organization:yarra-valley-water OR organization:yarriambiack-shire-council) -(organization:australian-institute-of-marine-science OR organization:geoscienceaustralia OR organization:csiro-oceans-and-atmosphere-information-and-data-centre OR organization:csiro-marine-and-amospheric-research OR organization:state-of-the-environment OR organization:new-south-wales-datasets OR organization:south-australian-governments)
```

**Comment by t83714** [comment]
This is the relevant issue that may impact item one (`open national map button` on DAG):
https://github.com/TerriaJS/terriajs/issues/4880

**Comment by soyarsauce** [comment]
Related: https://github.com/TerriaJS/terriajs/issues/4904

**Comment by mwu2018** [comment]
This ticket can be closed when all of the following issues are resolved:

- [ ] https://github.com/magda-io/magda/issues/3019

- [ ] https://github.com/TerriaJS/terriajs/issues/4906

- [ ] https://github.com/TerriaJS/TerriaMap/issues/494 (Also need to update saas fork)

- [ ] https://github.com/TerriaJS/terriajs/issues/4904

**Comment by mwu2018** [comment]
Upload the following json to https://nationalmap.gov.au and see the harvested ckan catalog without organisation filetering:

```
  {
    "services": [
      
    ],
    "homeCamera": {
      "north": -8,
      "east": 158,
      "south": -45,
      "west": 109
    },
    "corsDomains": [
      "corsproxy.com",
      "programs.communications.gov.au",
      "mapsengine.google.com",
      "s3-ap-southeast-2.amazonaws.com",
      "data.melbourne.vic.gov.au",
      "data.act.gov.au",
      "dea.ga.gov.au",
      "gsky.nci.org.au",
      "discover.data.vic.gov.au"
    ],
    "catalog": [
      {
        "blacklist": {
          "Rainforests (EVC_2005)": true
        },
        "groupBy": "organization",
        "minimumMaxScaleDenominator": 10000000000,
        "filterByWmsGetCapabilities": false,
        "filterQuery": [
          {
            "fq": "+(res_format:(geojson OR GeoJSON OR kml OR KML OR kmz OR KMZ OR wms OR WMS OR CSV-GEO-AU OR csv-geo-au OR aus-geo-csv OR \"Esri REST\"))"
          }
        ],
        "url": "http://www.data.gov.au",
        "type": "ckan",
        "name": "Data.gov.au (No org filtering)",
        "includeCsv": true,
        "includeGeoJson": true,
        "includeKml": true,
        "includeWms": true,
        "includeWfs": true,
        "includeEsriMapServer": true,
        "includeEsriFeatureServer": true
      }
    ]
  }

```

**Comment by mwu2018** [comment]
Without organisation filtering, the number of harvested ckan data items is substantially larger than with filtering:

![image](https://user-images.githubusercontent.com/41275384/101428569-1ad37080-3955-11eb-923d-2a43f000aa74.png)

**Comment by AnaBelgun** [comment]
@mwu2018 thank you; that looks great. Is that a way to check what's under each group? can I test it somewhere? maybe the data from councils coming through is not usable.

**Comment by AnaBelgun** [comment]
@mwu2018 
to continue investigation with: 

- [ ] the issue of inconsistent search results based on quality ratings
- [ ] the need for a UI to accommodate magda backed search of dga datasets harvested
